using-multifactor-authentication.md

```diff
@@ -0,0 +1,135 @@
+code. Use your authenticator to scan the QR-code. A new item will be shown as
+soon as the scan succeeds. If your authenticator cannot scan from camera, you
+can choose to manually add a new item. Fill information listed after the
+QR-code. Be cautious to type the _key_ correctly and choose _time based_
+option, and click _save_. You can see a 6-digit code changing with time (30
+seconds) in your app. Type the code and click _enable_.
+
+If the code is correct and QR-code not expired, you can see a list of recovery
+codes. Please keep them safe. Any of these codes can be used in place of digit
+code generated from authenticator, but every one can only be used once. So if
+your phone is missing or you unexpectedly deleted the account in the
+authenticator, you can use recovery code to disable multi-factor authentication
+first and re-generate a new key.
+
+## Authentication levels
+
+Back to _profile edit_ page, in _multifactor authentication_ section, you
+should see a dropdown menu with three items, about level of your
+authentication.
+
+- _disabled_, setting to this level is equivalent to disable your MFA.
+- _login only_, only login from browser or command line requires digit code.
+- _login and write_, login, pushing gems or changing gem ownership requires
+digit code.
+
+All level change requires authentication from digit code, or recovery code.
+To clear your multifactor authentication settings, choose _disabled_ and
+continue.
+
+## Using multi-factor authentication in command line
+
+<em class="t-gray">Support for MFA has not been in release version of RubyGems
+yet. Content below may be not suitable for your RubyGems installation.</em>
+
+Currently, the `gem` command line tool will check digit code in `gem signin`.
+`gem push` and `gem owner --add` or `gem owner --remove` would also require
+verification if your authentication level is set to _login and write_.
+
+If multi-factor authentication is enabled, `gem signin` will show a prompt
+asking digit code.
+
+    $ gem signin
+    Enter your RubyGems.org credentials.
+    Don't have an account yet? Create one at https://rubygems.org/sign_up
+    Email:   gem_author@example
+    Password:
+    
+    This command needs digit code for multifactor authentication.
+    Code:   111111
+    Signed in.
+
+There's a command line option for providing the digit code before execution.
+
+    $ gem signin --mfa 111111
+    Enter your RubyGems.org credentials.
+    Don't have an account yet? Create one at https://rubygems.org/sign_up
+    Email:   gem_author@example
+    Password:
+    
+    Signed in.
+
+Likewise, `gem push` and `gem owner` requires the code if your authentication
+level is set to _login and write_.
+
+    $ gem push hello-0.0.1.gem
+    Pushing gem to https://rubygems.org...
+    This command needs digit code for multifactor authentication.
+    Code:   111111
+    Successfully registered gem: hello (0.0.1)
+
+If your input is incorrect, the operation would fail.
```

Comment:
Just leave a not about otp expiring in 30 sec, we don't need to show example of incorrect input.